x-pack/plugins/ml/public/ui_actions/open_in_single_metric_viewer_action.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TimeRange } from '@kbn/es-query';
+import { i18n } from '@kbn/i18n';
+import { isPopulatedObject } from '@kbn/ml-is-populated-object';
+import { type EmbeddableApiContext, apiIsOfType } from '@kbn/presentation-publishing';
+import {
+  type UiActionsActionDefinition,
+  IncompatibleActionError,
+} from '@kbn/ui-actions-plugin/public';
+import { ML_APP_LOCATOR, ML_PAGES } from '../../common/constants/locator';
+import type { MlEmbeddableBaseApi, SingleMetricViewerEmbeddableApi } from '../embeddables';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '../embeddables';
+
+import type { MlCoreSetup } from '../plugin';
+
+export interface OpenInSingleMetricViewerActionContext extends EmbeddableApiContext {
+  embeddable: SingleMetricViewerEmbeddableApi;
+}
+
+export const OPEN_IN_SINGLE_METRIC_VIEWER_ACTION = 'openInSingleMetricViewerAction';
+
+export function isSingleMetricViewerEmbeddableContext(
+  arg: unknown
+): arg is OpenInSingleMetricViewerActionContext {
+  return (
+    isPopulatedObject(arg, ['embeddable']) &&
+    apiIsOfType(arg.embeddable, ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE)
+  );
+}
+
+const getTimeRange = (embeddable: MlEmbeddableBaseApi): TimeRange | undefined => {
+  return embeddable.timeRange$?.getValue() ?? embeddable.parentApi?.timeRange$?.getValue();
+};
+
+export function createOpenInSingleMetricViewerAction(
+  getStartServices: MlCoreSetup['getStartServices']
+): UiActionsActionDefinition<OpenInSingleMetricViewerActionContext> {
+  return {
+    id: 'open-in-single-metric-viewer',
+    type: OPEN_IN_SINGLE_METRIC_VIEWER_ACTION,
```

Comment:
We need to set the `order` to position this action higher in the context menu 
```suggestion
    type: OPEN_IN_SINGLE_METRIC_VIEWER_ACTION,
    order: 100,
```

Reply:
Updated in 902197b87cbc099790e53f64e81b6bc19fc60ec4
Screen shot in description has also been updated to reflect the change.